resources/2025-11-11-secrets-redactdata.md

```diff
@@ -0,0 +1,477 @@
+# Redacting Sensitive Data for Radius.Security/secrets
+
+* **Author**: Lakshmi Javadekar (@lakshmimsft)
+
+## Overview
+
+The `Radius.Security/secrets` resource type, defined in the [resource-types-contrib](https://github.com/radius-project/resource-types-contrib) repository, enables users to deploy sensitive data such as tokens, passwords, keys, and certificates to various secret backends (Azure Key Vault, HashiCorp Vault, Kubernetes secrets, etc.). This resource is handled by dynamic-rp, a type-agnostic processor providing CRUD operations for user-defined types.
+
+Currently, dynamic-rp stores all resource properties, including sensitive `data` fields, in the Radius database. This design proposes a mechanism to redact (nullify) sensitive data after the recipe successfully deploys the secrets to their designated backend, ensuring that secret data does not persist in the Radius database.
+
+## Terms and definitions
+
+| Term | Definition |
+|------|------------|
+| **Dynamic-RP** | Type-agnostic resource provider that handles CRUD operations for user-defined types without prior knowledge of their schemas |
+| **Recipe** | Deployment template (Bicep/Terraform) that provisions infrastructure resources |
+| **Recipe-based deployment** | Asynchronous deployment flow where recipes execute to create actual infrastructure |
+| **Radius.Security/secrets** | New resource type for managing secrets across multiple backends (recipe-based, multi-backend) |
+
+## Objectives
+
+> **Issue Reference:** Implementation of sensitive data handling for Radius.Security/secrets resource type to prevent long-term storage of plaintext secrets in Radius database. ref: https://github.com/radius-project/radius/issues/10421
+
+### Goals
```

Comment:
can you talk a bit about the work here enabling use cases where `Radius.Security/secrets` can be referenced by other recipes to read and use secret values? e.g. containers recipe connecting to a secrets resource for mounting the secrets as a volume and reading the secret value to inject env vars, etc.